db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/SessionPropertiesImpl.java

```diff
@@ -104,11 +111,59 @@ public void setProperty(PROPERTYNAME property, Object value) {
                 boolean useNativeSpark = Boolean.parseBoolean(valString);
                 properties[USE_NATIVE_SPARK.getId()] = useNativeSpark;
                 break;
+            case CURRENTFUNCTIONPATH:
+                properties[CURRENTFUNCTIONPATH.getId()] = parseCurrentFunctionPath(valString);
+                break;
             default:
                 assert false;
         }
     }
 
+    /**
+     * parses the value of CURRENTFUNCTIONPATH according to:
+     * https://www.ibm.com/support/producthub/db2/docs/content/SSEPGG_11.5.0/com.ibm.db2.luw.apdv.cli.doc/doc/r0008777.html
+     * @param value the user-input value
+     * @return list of schemas for function lookup.
+     */
+    private String[] parseCurrentFunctionPath(String value) {
+        if(value == null || value.isEmpty()) {
+            return new String[] {"SYSIBM", "SYSFUN"};
+        } else {
+            // source: https://stackabuse.com/regex-splitting-by-character-unless-in-quotes/
+            List<String> candidateSchemas = new ArrayList<>(Arrays.asList(value.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)")));
+            for(int i = 0; i < candidateSchemas.size(); ++i) {
+                candidateSchemas.set(i , normalizeAndUnquoteIdentifier(candidateSchemas.get(i)));
+            }
+            boolean sysibmSeen = false;
+            boolean sysfunSeen = false;
+            for(int i = 0; i < candidateSchemas.size(); ++i) {
+                String candidateSchema = candidateSchemas.get(i);
+                if(i == 0 && candidateSchema.equals("SYSIBM")) {
+                    sysibmSeen = true;
+                } else if(i == 1 && candidateSchema.equals("SYSFUN")) {
+                    sysfunSeen = true;
+                } else {
+                    break;
+                }
+            }
+            if(!sysibmSeen) {
+                candidateSchemas.add(0, "SYSIBM");
+            }
+            if(!sysfunSeen) {
+                candidateSchemas.add(1, "SYSFUN");
+            }
+            return candidateSchemas.toArray(new String[0]);
+        }
+    }
+
+    private String normalizeAndUnquoteIdentifier(String identifier) {
+        String normalized = StringUtil.normalizeSQLIdentifier(identifier);
+        if(!normalized.isEmpty() && normalized.charAt(0) == '"' && normalized.charAt(normalized.length() - 1) == '"') {
```

Comment:
another possibility, not sure if it work faster, but looks shorter:)
normalized.replaceAll("^\"|\"$", "");

Reply:
Interesting idea, I thought about this and I think it allows invalid schemas such as `"FOO` to be converted to `FOO` which is not correct, this is because of the OR in the Regex `"^\\\"|\\\"$"`, i.e.:
```
String normalized = "hello world\"";
System.out.println(normalized.replaceAll("^\\\"|\\\"$", ""));
// returns: hello world
```
To fix this, we could maybe use Regex's `()()`'s AND syntax as [explained here](https://stackoverflow.com/questions/6185438/and-operator-in-regular-expressions) to achieve this, then we end up with something like this `normalized.matches("(?=.*^\\\")(?=.*\\\"$).*")` which honestly looks a bit more clumsy than current version :D 

Let me know if there is a better way to do this

Reply:
sorry, I missed the fact, that it replaces only when quotes are on both sides. Then let's keep so, as it easier to read compare to regex:)